src/actions/datafiles.js

```diff
@@ -38,7 +38,7 @@ export function fetchDataFile(filename) {
   };
 }
 
-export function putDataFile(filename, data, source = "editor") {
+export function putDataFile(filename, data, source = "editor", ext = "") {
```

Comment:
Why not get the extension from the filename? We have a [helper](https://github.com/jekyll/jekyll-admin/blob/master/src/utils/helpers.js#L67) for that too.